controllers/admin/AdminThemesController.php

```diff
@@ -1,1120 +0,0 @@
- * 2007-2018 PrestaShop.
- *
- * NOTICE OF LICENSE
- *
- * This source file is subject to the Open Software License (OSL 3.0)
- * that is bundled with this package in the file LICENSE.txt.
- * It is also available through the world-wide-web at this URL:
- * https://opensource.org/licenses/OSL-3.0
- * If you did not receive a copy of the license and are unable to
- * obtain it through the world-wide-web, please send an email
- * to license@prestashop.com so we can send you a copy immediately.
- *
- * DISCLAIMER
- *
- * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
- * versions in the future. If you wish to customize PrestaShop for your
- * needs please refer to http://www.prestashop.com for more information.
- *
- * @author    PrestaShop SA <contact@prestashop.com>
- * @copyright 2007-2018 PrestaShop SA
- * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
- * International Registered Trademark & Property of PrestaShop SA
- */
-use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManager;
-use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManagerBuilder;
-use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeRepository;
-use PrestaShop\PrestaShop\Core\Shop\LogoUploader;
-use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
-
-/**
- * @property Theme $object
- */
-class AdminThemesControllerCore extends AdminController
-{
-    // temp
-    const CACHE_FILE_CUSTOMER_THEMES_LIST = '/config/xml/customer_themes_list.xml';
-    const CACHE_FILE_MUST_HAVE_THEMES_LIST = '/config/xml/must_have_themes_list.xml';
-
-    /**
-     * @var ThemeManager
-     */
-    public $theme_manager;
-
-    /**
-     * @var ThemeRepository
-     */
-    protected $theme_repository;
-    protected $toolbar_scroll = false;
-    protected $authAccesses = array();
-    private $img_error;
-
-    /* @var LogoUploader $logo_uploader */
-    private $logo_uploader;
-
-    public function __construct()
-    {
-        $this->bootstrap = true;
-        parent::__construct();
-
-        $this->theme_manager = (new ThemeManagerBuilder($this->context, Db::getInstance()))->build();
-        $this->theme_repository = (new ThemeManagerBuilder($this->context, Db::getInstance()))->buildRepository();
-        $this->logo_uploader = new LogoUploader($this->context->shop);
-    }
-
-    public function init()
-    {
-        // No cache for auto-refresh uploaded logo
-        header('Cache-Control: no-cache, must-revalidate');
-        header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
-
-        parent::init();
-        $this->can_display_themes = (!Shop::isFeatureActive() || Shop::getContext() == Shop::CONTEXT_SHOP);
-
-        if (Tools::getValue('display') == 'configureLayouts') {
-            $this->initConfigureLayouts();
-        } elseif (Tools::getValue('action') == 'importtheme') {
-            $this->display = 'importtheme';
-        }
-
-        libxml_use_internal_errors(true);
-    }
-
-    public function downloadAddonsThemes()
```

Comment:
@matks i did not migrate `Download themes from addons when employee is logged in into Addons` feature because it does not work. :/

Reply:
😮 temporary or totally broken ?

Reply:
it was totally broken, so doesnt even work. i guess its since 1.7.0.0.